File: flapypan-blog-api/src/main/java/top/flapypan/blog/service/LinkService.java
package top.flapypan.blog.service;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import top.flapypan.blog.entity.Link;
import top.flapypan.blog.repository.LinkRepository;

import java.util.List;

@Slf4j
@RequiredArgsConstructor
@Service
public class LinkService {

    private final LinkRepository repository;

    public List<Link> getAll() {
        return repository.findAll();
    }

    @Transactional
    public List<Link> save(List<Link> links) {
        repository.deleteAll();
        return repository.saveAll(links);
    }
}

File: flapypan-blog-api/src/main/kotlin/top/flapypan/blog/service/LinkService.kt
package top.flapypan.blog.service

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import top.flapypan.blog.entity.Link
import top.flapypan.blog.repository.LinkRepository

/**
 * 固定链接相关服务
 */
@Service
class LinkService(
    private val repository: LinkRepository
) {

    /**
     * 获取所有固定链接
     */
    fun getAll(): List<Link> = repository.findAll()

    /**
     * 保存固定链接
     */
    @Transactional
    fun save(links: List<Link>): List<Link> {
        repository.deleteAll()
        return repository.saveAll(links)
    }
}
